traffic_ops/testing/api/v4/cachegroupsdeliveryservices_test.go

```diff
@@ -20,143 +20,88 @@ import (
-	}
+	assert.RequireNoError(t, err, "Cannot GET DeliveryServices: %v - %v", err, dses)
 
 	opts := client.NewRequestOptions()
-	opts.QueryParameters.Set("name", TestEdgeServerCacheGroupName)
+	opts.QueryParameters.Set("name", "cachegroup3")
 	clientCGs, _, err := TOSession.GetCacheGroups(opts)
-	if err != nil {
-		t.Fatalf("getting cachegroup: %v", err)
-	}
-	if len(clientCGs.Response) != 1 {
-		t.Fatalf("getting cachegroup expected 1, got %v", len(clientCGs.Response))
-	}
-
-	clientCG := clientCGs.Response[0]
-
-	if clientCG.ID == nil {
-		t.Fatalf("Cachegroup has a nil ID")
-	}
-	cgID := *clientCG.ID
+	assert.RequireNoError(t, err, "Cannot GET cachegroup: %v", err)
+	assert.RequireEqual(t, len(clientCGs.Response), 1, "Getting cachegroup expected 1, got %v", len(clientCGs.Response))
+	assert.RequireNotNil(t, clientCGs.Response[0].ID, "Cachegroup has a nil ID")
 
 	dsIDs := []int{}
-	topologyDsIDs := []int{}
 	for _, ds := range dses.Response {
 		if *ds.CDNName == "cdn1" && ds.Topology == nil {
 			dsIDs = append(dsIDs, *ds.ID)
-		} else if *ds.CDNName == "cdn1" && ds.Topology != nil {
-			topologyDsIDs = append(topologyDsIDs, *ds.ID)
-		}
-	}
-	if len(dsIDs) < 1 {
-		t.Fatal("No Delivery Services found in CDN 'cdn1', cannot continue.")
-	}
-
-	if len(topologyDsIDs) < 1 {
-		t.Fatal("No Topology-based Delivery Services found in CDN 'cdn1', cannot continue.")
-	}
-
-	_, reqInf, err := TOSession.SetCacheGroupDeliveryServices(cgID, topologyDsIDs, client.RequestOptions{})
-	if err == nil {
-		t.Fatal("assigning Topology-based delivery service to cachegroup - expected: error, actual: nil")
-	}
-	if reqInf.StatusCode < http.StatusBadRequest || reqInf.StatusCode >= http.StatusInternalServerError {
-		t.Fatalf("assigning Topology-based delivery service to cachegroup - expected: 400-level status code, actual: %d", reqInf.StatusCode)
-	}
-
-	resp, _, err := TOSession.SetCacheGroupDeliveryServices(cgID, dsIDs, client.RequestOptions{})
-	if err != nil {
-		t.Fatalf("setting cachegroup delivery services returned error: %v", err)
-	}
-	if len(resp.Response.ServerNames) == 0 {
-		t.Fatal("setting cachegroup delivery services returned success, but no servers set")
-	}
-
-	// Note this second post of the same cg-dses specifically tests a previous bug, where the query
-	// failed if any servers with location parameters were already assigned, due to a foreign key
-	// violation. See https://github.com/apache/trafficcontrol/pull/3199
-	resp, _, err = TOSession.SetCacheGroupDeliveryServices(cgID, dsIDs, client.RequestOptions{})
-	if err != nil {
-		t.Fatalf("setting cachegroup delivery services returned error: %v", err)
-	}
-	if len(resp.Response.ServerNames) == 0 {
-		t.Fatal("setting cachegroup delivery services returned success, but no servers set")
-	}
-
-	opts.QueryParameters.Del("name")
-	for _, serverName := range resp.Response.ServerNames {
-		opts.QueryParameters.Set("hostName", string(serverName))
```

Comment:
Where is this check happening?

Reply:
Hm, yeah this is validating that the servers belonging to the cachegroup are now assigned to the delivery service. I missed this, I will add the validation function.